### PR TITLE
Update impl to CANTO lending

### PR DESCRIPTION
**Pull request by jessepinkman9900**

### CODE REVIEW

File: abis/Comptroller.json [hunk 1]
```diff
@@ -2157,4 +2157,4 @@
     "stateMutability": "nonpayable",
     "type": "function"
   }
-]
\ No newline at end of file
+]
```
Comment (zjesko) [inline abis/Comptroller.json:2160]: Is this change required?

File: schema.graphql [hunk 2]
```diff
@@ -33,8 +31,18 @@ type Market @entity { # TODO rename to CToken, not market AND ALL OTHERS
...
+    "supply APY"
+    supplyAPY: BigDecimal!
+    "supply distribution APY"
+    supplyDistributionAPY: BigDecimal!
+    "Yearly borrow rate. With x blocks per year at 5 s/block"
+    borrowRate: BigDecimal!
+    "borrow APY"
+    borrowAPY: BigDecimal!
+    "borrowDistribution APY"
+    borrowDistributionAPY: BigDecimal!
```
Comment (zjesko) [inline schema.graphql:47]: borrowDistributionAPY is not needed, lets remove it

File: schema.graphql [hunk 3]
```diff
@@ -33,8 +31,18 @@ type Market @entity { # TODO rename to CToken, not market AND ALL OTHERS
     numberOfSuppliers: Int!
     "Reserves stored in the contract"
     reserves: BigDecimal!
-    "Yearly supply rate. With 2104400 blocks per year"
+    "Yearly supply rate. With x blocks per year at 5 s/block"
     supplyRate: BigDecimal!
+    "supply APY"
+    supplyAPY: BigDecimal!
```
Comment (zjesko) [inline schema.graphql:39]: what is the difference between `supplyRate` and `supplyAPY` (same for borrowRate and borrowAPY). Are both needed if one can be derived from the other?

File: src/mappings/comptroller.ts [hunk 4]
```diff
@@ -1,20 +1,28 @@
...
 
 import { Market, Comptroller } from '../types/schema'
-import { mantissaFactorBD, updateCommonCTokenStats } from './helpers'
+import { MANTISSA_FACTOR_BD } from './consts'
+import { updateCommonCTokenStats } from './helpers'
+import { createMarket } from './markets'
 
 export function handleMarketEntered(event: MarketEntered): void {
-  let market = Market.load(event.params.cToken.toHexString())
+  log.info('COMPTROLLER::handleMarketEntered', [])
```
Comment (zjesko) [inline src/mappings/comptroller.ts]: please remove the redundant logging statements. If they are for debugging, don't include them in the PR

File: src/mappings/comptroller.ts [hunk 5]
```diff
@@ -29,7 +37,12 @@ export function handleMarketEntered(event: MarketEntered): void {
 }
 
 export function handleMarketExited(event: MarketExited): void {
-  let market = Market.load(event.params.cToken.toHexString())
+  log.info('COMPTROLLER::handleMarketExited', [])
+  let marketId = event.params.cToken.toHexString()
+  let market = Market.load(marketId)
+  if (market === null) {
```
Comment (zjesko) [inline src/mappings/comptroller.ts:41]: why are we adding this extra `market==null` check everywhere, it is always not null right (acc to compound code). 
Did we have any instance where market was null and it was breaking?

File: src/mappings/comptroller.ts [hunk 6]
```diff
@@ -44,33 +57,41 @@ export function handleMarketExited(event: MarketExited): void {
...
+  log.info('COMPTROLLER::handleNewCollateralFactor', [])
+  let marketId = event.params.cToken.toHex()
+  let market = Market.load(marketId)
+  if (market == null) {
+    market = createMarket(marketId)
+  }
   market.collateralFactor = event.params.newCollateralFactorMantissa
     .toBigDecimal()
-    .div(mantissaFactorBD)
+    .div(MANTISSA_FACTOR_BD)
```
Comment (zjesko) [inline src/mappings/comptroller.ts:71]: lets avoid renaming variables unless absolutely needed as we are taking changes to a fork and not writing it from scratch? wdyt?

File: src/mappings/markets.ts [hunk 7]
```diff
@@ -49,125 +81,316 @@ function getTokenPrice(
...
+  if (!result.reverted) {
+    symbol = result.value
+  } else {
+    log.info('CUSTOM' + address.toString(), [])
+  }
+
+  return symbol
+}
+
+function cToken_name(marketAddress: string, contract: CToken): string {
```
Comment (zjesko) [inline src/mappings/markets.ts:186]: let remove this hardcoding for name, symbol, decimals etc

File: src/mappings/tokenDefinition.ts [hunk 8]
```diff
@@ -0,0 +1,147 @@
+import { Address, BigInt } from '@graphprotocol/graph-ts'
```
Comment (zjesko) [inline src/mappings/tokenDefinition.ts]: do we need these tokenDefinitions when we can directly call the contract to avoid as much hardcoding as possible?

File: subgraph.yaml [hunk 9]
```diff
@@ -499,38 +481,3 @@ dataSources:
           handler: handleTransfer
         - event: NewMarketInterestRateModel(address,address)
           handler: handleNewMarketInterestRateModel
-  - kind: ethereum/contract
```
Comment (zjesko) [inline subgraph.yaml]: why did we change the positioning of comptroller to the end of the file from the beginning?

File: src/mappings/comptroller.ts (hunk 5 above)
Comment (jessepinkman9900) [reply src/mappings/comptroller.ts:41]: yes, there were some instance where it was coming back as null and breaking

File: src/mappings/comptroller.ts (hunk 4 above)
Comment (jessepinkman9900) [reply src/mappings/comptroller.ts]: removed the logging statements

File: schema.graphql (hunk 3 above)
Comment (jessepinkman9900) [reply schema.graphql:39]: rate is like simple interest used in APY calculation

File: schema.graphql (hunk 2 above)
Comment (jessepinkman9900) [reply schema.graphql:47]: - compound v3 dash shows distAPY - https://v3-app.compound.finance/markets

File: src/mappings/comptroller.ts (hunk 6 above)
Comment (jessepinkman9900) [reply src/mappings/comptroller.ts:71]: I did it for easier code readability. consts def using caps

File: subgraph.yaml (hunk 9 above)
Comment (jessepinkman9900) [reply subgraph.yaml]: moved it so that comptroller is at top similar to how factory is at top in dex subgraph. resolved. moved it back to the bottom

**Comment by jessepinkman9900** [comment]
TODO: update naming in front-end as cToken names are inconsistent in smart contract

File: src/mappings/tokenDefinition.ts (hunk 8 above)
Comment (jessepinkman9900) [reply src/mappings/tokenDefinition.ts]: resolved

File: src/mappings/markets.ts (hunk 7 above)
Comment (jessepinkman9900) [reply src/mappings/markets.ts:186]: resolved

**Comment by jessepinkman9900** [comment]
@akshatcx @SiddarthVijay please review

**Review by zjesko — APPROVED** [review]
